api/applications/applications_handler.go

```diff
@@ -277,13 +291,20 @@ func (ah *ApplicationHandler) ModifyRegistrationDetails(ctx context.Context, app
 		return nil, err
 	}
 
-	payload := []patch{}
+	payload := make([]patch, 0)
 	runUpdate := false
 	updatedRegistration := currentRegistration.DeepCopy()
 
 	// Only these fields can change over time
 	patchRequest := applicationRegistrationPatchRequest.ApplicationRegistrationPatch
 	if patchRequest.AdGroups != nil && !radixutils.ArrayEqualElements(currentRegistration.Spec.AdGroups, *patchRequest.AdGroups) {
+		valid, err := ah.validateUserIsMemberOfAdGroups(ctx, appName, patchRequest.AdGroups)
+		if err != nil {
+			return nil, err
+		}
+		if !valid {
+			return nil, radixhttp.ValidationError("Radix Registration", "User should be a member of at least one admin AD group or their sub-members")
+		}
```

Comment:
Move this validation into the `if runUpdate` block and use `updatedRegistration.Spec.AdGroups` as argument to `ah.validateUserIsMemberOfAdGroups`. Also only run validation if length of AdGroups > 0

Reply:
Fixed